airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcDatabase.java

```diff
@@ -116,4 +124,30 @@ default void executeWithinTransaction(List<String> queries) throws SQLException
                       CheckedFunction<ResultSet, T, SQLException> recordTransform)
       throws SQLException;
 
+  default <T> Optional<T> querySingle(String sql, CheckedFunction<ResultSet, T, SQLException> recordTransform, String... params)
```

Comment:
would it make sense to call this `executeParameterized` and have it return a `Stream<T>`? after that you can still do a `queryFirst` or just in the consuming code fo the `findFirst`

Reply:
if you do that you might be able to use it in `listConfigs`.